### PR TITLE
Closes #55

### LINKED ISSUE #55: `GeneralModel` class clean up
To do:

- [ ] Revise documentation
  - [ ] RStudio outline section dividers (use `# function or class name ----`)
  - [ ] Add lifecycle badge (as a default take "stable" for existing functions or classes)
  - [ ] Replace `##'` by `#'`
  - [ ] Add to `pkgdown.yml`
  - [ ] Remove file history and the timestamp in comments on top of file
- [ ] Add unit tests
  - [ ] For classes:
    - [ ] Test default constructor
    - [ ] For reference classes: test methods
  - [ ] For functions: as usual
- [ ] Refactor as needed
  - [ ] Code styling (indentation etc.) - install package `styler` and use corresponding add-in
  - [ ] Keep externally used names so that we don't break downstream code
  - [ ] Remove `keywords` tags
  - [ ] For functions or methods:
    - [ ] Add assertions for arguments using `checkmate` functions
    - [ ] If it is too long, break up into pieces (rule of thumb: maximum 50 lines of code)
    - [ ] Consider renaming internal variables etc. if needed
    - [ ] Avoid explicit use of `return`
  - [ ] For classes:
    - [ ] move function definitions outside if possible (e.g. move validation function as `validity_subjects()` to `Data-validity.R` and have corresponding test file `test-Data-validity.R`)
    - [ ] In `setClass` function, just use the `slots` instead of outdated `representation`
    - [ ] Remove things like `validObject(.Data())`

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #55

### CODE REVIEW

File: tests/testthat/test-Model-validity.R [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      x
+    },
+    modelspecs = function(x) {
+      x
+    },
+    init = function(x) {
+      x
+    },
+    sample = "param1"
+  )
```
Comment (danielinteractive) [inline tests/testthat/test-Model-validity.R:19]: maybe put this in another test helper function that gives you a `GeneralModel` object back?

**Review by danielinteractive — APPROVED** [review]
thx Wojciech, only minor comments

File: R/Model-class.R [hunk 2]
```diff
@@ -1,99 +1,73 @@
...
+  Class = "GeneralModel",
+  slots = c(
+    datamodel = "function",
+    priormodel = "function",
+    modelspecs = "function",
+    init = "function",
+    sample = "character"
+  ),
+  contains = "AllModels",
+  validity = validate_model_general
```
[the remark this answers is not in the document]
Comment (wwojciech) [reply R/Model-class.R:67]: Sure, actually, I had it like you propose and then I changed :)

File: tests/testthat/test-Model-validity.R (hunk 1 above)
Comment (wwojciech) [reply tests/testthat/test-Model-validity.R:19]: Sure, but I think these 2 places would be only the one we need it, and e.g. for .GeneralData we did not create such helper. Let me know if you still prefer do make it :)

**Pull request merged**